bot/cogs/antimalware.py

```diff
@@ -18,6 +18,10 @@ def __init__(self, bot: Bot):
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         """Identify messages with prohibited attachments."""
+        # Check if user is staff, if is, return
+        if type(message.author) is Member and any(role.id in STAFF_ROLES for role in message.author.roles):
```

Comment:
It's more efficient if the roles are checked after the check for existence of attachments (line 25). Furthermore, at the least, use `isinstance()` rather than `type() is ...`. However, I think checking if the attribute exists is even better:

```suggestion
        if hasattr(message.author, "roles") and any(role.id in STAFF_ROLES for role in message.author.roles):
```

That being said, this should probably ignore DMs (`if not message.guild`). If that happens, then it may be safe to assume the author is always a `Member` but I am not certain of that. It doesn't hurt to be cautious and explicitly check they have roles.